db/migrate/20160415231343_clean_duplicate_reservations.rb

```diff
@@ -0,0 +1,27 @@
+class CleanDuplicateReservations < ActiveRecord::Migration
+  class Reservation < ActiveRecord::Base
+  end
+
+  def up
+    order_details_with_multiple_reservations = Reservation
+      .where("order_detail_id is not null")
+      .group(:order_detail_id)
+      .having("count(*) > 1")
+      .pluck(:order_detail_id)
+
+    order_details_with_multiple_reservations.each do |order_detail_id|
+      Reservation.where(order_detail_id: order_detail_id).last.destroy
+    end
+    if NUCore::Database.oracle?
+      remove_index :reservations, :order_detail_id
```

Comment:
I realize this is awkward. MySQL ties foreign keys very tightly to an index. So I can't remove this index without also removing the foreign key.  Oracle keeps the foreign keys and indexes completely separate so I can actually replace the index.